test/src/test/java/hudson/tasks/ShellTest.java

```diff
@@ -68,10 +73,83 @@ public Proc onLaunch(ProcStarter p) throws IOException {
         FreeStyleProject p = rule.createFreeStyleProject();
         p.getBuildersList().add(new Shell("echo abc"));
         p.setAssignedNode(s);
-        
+
         FreeStyleBuild b = rule.assertBuildStatusSuccess(p.scheduleBuild2(0).get());
 
         assertEquals(1,s.numLaunch);
         assertTrue(FileUtils.readFileToString(b.getLogFile()).contains("Hudson was here"));
     }
+
+    /* A FakeLauncher that just returns the specified error code */
+    private class ReturnCodeFakeLauncher implements FakeLauncher {
+        final int code;
+
+        ReturnCodeFakeLauncher(int code)
+        {
+            super();
+            this.code = code;
+        }
+
+        @Override
+        public Proc onLaunch(ProcStarter p) throws IOException {
+            return new FinishedProc(this.code);
+        }
+    }
+
+    private static Shell createNewShell(String command, Integer unstableReturn) {
+        Shell shell = new Shell(command);
+        shell.setUnstableReturn(unstableReturn);
+        return shell;
+    }
+
+    @Issue("JENKINS-23786")
+    public void testUnstableReturn() throws Exception {
+        if(Functions.isWindows())
```

Comment:
:ant: Should be <code>assumeThat</code> in order to mark the test as skipped instead of passed.